enhancements/network/ovn-kubernetes-ns-ipsec.md

```diff
@@ -0,0 +1,185 @@
+  - "/enhancements/network/ovn-kubernetes-ipsec.md"
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# OVN Kubernetes North-South IPsec
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The OVN Kubernetes network plugin uses [OVN](https://www.ovn.org) to instantiate
+virtual networks for Kubernetes. These virtual networks use Geneve, a network
+virtualization overlay encapsulation protocol, to tunnel traffic across the
+underlay network between Kubernetes Nodes.
+
+IPsec is a protocol suite that enables secure network communications between
+IP endpoints by providing the following services:
+- Confidentiality
+- Authentication
+- Data Integrity
+
+OVN tunnel traffic is transported by physical routers and switches. These
+physical devices could be untrusted (devices in public network) or might be
+compromised. Enabling IPsec encryption for this tunnel traffic can prevent the
+traffic data from being monitored and manipulated.
+
+The scope of this work is to enable customers to encrypt traffic going in/out of a node. It will not encrypt traffic between cluster nodes (for that we have ovn-kubernetes-ipsec.md). However due the fact that both will use the same mechanims, n-s implementation will also touch the existing e-w.
+
+## Motivation
+
+Encryption services are recommended when traffic is traversing an untrusted
+network. Encryption services may also be required for regulatory or compliance
+reasons (e.g. FIPS compliance, FSI regulations, and more).
+
+Use-Cases to consider:
+a. Encapsulate all traffic - when cluster is placed remotely and need to communicate with services in a central DC (cloud or other), and last mile of communication is not controlled by owner (ie - public internet), we'd want to encapsulate all traffic in IPsec tunnel (as VPN). this includes even ssh traffic etc. cluster can be placed behind NAT and we might even want to encapsulate IPv6 inside IPv4 tunnel. this use case need to be available independently of kubelet.
+
+b. CSI encryption - some regulation demand encryption in-transit and at-rest for certain types of data. IPsec tunnel can be used to encrypt data in-transit for cases where we cant use anything else (aka NetApp ONTAP). this use case user a policy to encrypt only toward certain endpoints.
+
+
+### Goals
+
+- Provide per host north-south IPsec encryption that encapsulate all traffic.
+- Provide per host north-south IPsec encryption with parties external to the cluster.
+- Said IPsec should be "stand alone" and start before kubelet
+
+### Non-Goals
+
+- RHEL nodes, support will be for RHCOS only see section below on RHEL nodes
+- Shared Gateway Mode (SGW), only local GW mode will be supported
```

Comment:
@yuvalk   Can you help confirm this? IPSEC NS only for local GW mode?   cc @anuragthehatter